webapp/src/components/user_attribute/index.ts

```diff
@@ -0,0 +1,50 @@
+import {connect} from 'react-redux';
+import {AnyAction, Dispatch, bindActionCreators} from 'redux';
+
+import {UserProfile} from 'mattermost-redux/types/users';
+
+import {getGitlabUser} from '../../actions';
+
+import {GlobalState, pluginStateKey} from 'src/types/store';
+
+import UserAttribute from './user_attribute';
+
+interface UserAttributeStateProps {
+    id: string,
+    username: string,
+    gitlabURL: string,
+}
+
+interface UserAttributeDispatchProps {
+    actions: {
+        getGitlabUser:(userID: string) => (dispatch: Dispatch<AnyAction>, getState: () => GlobalState) => Promise<any>,
+    },
+}
+
+export type UserAttributeProps = UserAttributeStateProps & UserAttributeDispatchProps
```

Comment:
Can we move this up further in the file? Typescript allows you to specify types out of order